src/Kestrel.Transport.Sockets/SocketTransport.cs

```diff
@@ -112,23 +125,39 @@ public Task StopAsync()
             {
                 while (true)
                 {
-                    var acceptSocket = await _listenSocket.AcceptAsync();
-
-                    acceptSocket.NoDelay = _endPointInformation.NoDelay;
-
-                    var connection = new SocketConnection(acceptSocket, _pipeFactory, _trace);
-                    _ = connection.StartAsync(_handler);
+                    try
+                    {
+                        var acceptSocket = await _listenSocket.AcceptAsync();
+                        acceptSocket.NoDelay = _endPointInformation.NoDelay;
+
+                        var connection = new SocketConnection(acceptSocket, _pipeFactory, _trace);
+                        _ = connection.StartAsync(_handler);
+                    }
+                    catch (SocketException ex) when (ex.SocketErrorCode == SocketError.ConnectionReset)
+                    {
+                        // REVIEW: Should there be a seperate log message for a connection reset this early?
+                        _trace.ConnectionReset(connectionId: "(null)");
```

Comment:
Do you want to log connection reset when unbinding?

Reply:
Yes. AFAICT unbinding doesn't cause a SocketError.ConnectionReset. If it happens to occur during shutdown, I don't see the harm in this info level log message.